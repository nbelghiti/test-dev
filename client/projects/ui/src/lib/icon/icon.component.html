<i class="c-icon__glyph {{ size && 'c-icon__glyph--' + size }} {{ type && 'c-icon__glyph--' + type }} fas fa-{{ name }}"></i>
