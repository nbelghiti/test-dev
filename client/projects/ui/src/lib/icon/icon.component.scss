// =============================================================================
// :: Imports
// =============================================================================
@import "global";

// =============================================================================
// :: Icon
// =============================================================================
:host {
	display: inline-flex;
	justify-content: center;
	align-items: center;

	font-size: inherit;
}

// =============================================================================
// :: Elements
// =============================================================================
.c-icon__glyph {
	line-height: inherit;

	cursor: pointer;
}

// =============================================================================
// :: Modifiers
// =============================================================================
.c-icon__glyph--large {
	font-size: 2rem;
}

.c-icon__glyph--small {
	font-size: 1.4rem;
}

.c-icon__glyph--tiny {
	font-size: 1.2rem;
}

.c-icon__glyph--success {
	color: $ui-color-success-dark;
}

.c-icon__glyph--error {
	color: $ui-color-error-dark;
}

.c-icon__glyph--warning {
	color: $ui-color-warning-dark;
}

.c-icon__glyph--info {
	color: $ui-color-info-dark;
}
