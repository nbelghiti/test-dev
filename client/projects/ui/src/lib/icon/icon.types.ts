export enum IconName {
	archive = 'archive',
	arrowDown = 'arrow-down',
	arrowLeft = 'arrow-left',
	arrowRight = 'arrow-right',
	arrowRightAlt = 'arrow-alt-circle-right',
	arrows = 'arrows-alt-h',
	arrowUp = 'arrow-up',
	bell = 'bell',
	bellSlash = 'bell-slash',
	book = 'book',
	brackets = 'brackets',
	calendar = 'calendar',
	caretDown = 'caret-down',
	caretLeft = 'caret-left',
	caretRight = 'caret-right',
	caretUp = 'caret-up',
	check = 'check',
	checkCircle = 'check-circle',
	chevronDown = 'chevron-down',
	chevronLeft = 'chevron-left',
	chevronRight = 'chevron-right',
	chevronUp = 'chevron-up',
	clock = 'clock',
	clone = 'clone',
	close = 'times-circle',
	code = 'code',
	codeBranch = 'code-branch',
	cube = 'cube',
	download = 'download',
	edit = 'edit',
	excel = 'file-excel',
	exclamation = 'exclamation',
	exclamationCircle = 'exclamation-circle',
	expand = 'expand',
	externalLink = 'external-link-alt',
	file = 'file-alt',
	home = 'home',
	horizontalEllipsis = 'ellipsis-h',
	industry = 'industry',
	infoCircle = 'info-circle',
	invisible = 'eye-slash',
	marker = 'map-marker-alt',
	minus = 'minus',
	pdf = 'file-pdf',
	pen = 'pen',
	plus = 'plus',
	print = 'print',
	search = 'search',
	settings = 'cog',
	signOut = 'sign-out-alt',
	snowflake = 'snowflake',
	spinner = 'spinner',
	star = 'star',
	sync = 'sync-alt',
	table = 'table',
	times = 'times',
	trash = 'trash',
	upload = 'upload',
	user = 'user',
	users = 'users',
	visible = 'eye',
	plusSquare = 'plus-square',
	minusSquare = 'minus-square',
}

export enum IconType {
	success = 'success',
	error = 'error',
	warning = 'warning',
	info = 'info',
}

export enum IconSize {
	large = 'large',
	small = 'small',
	tiny = 'tiny',
}
