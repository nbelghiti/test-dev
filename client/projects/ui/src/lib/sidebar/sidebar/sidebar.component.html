<aside
	class="c-sidebar"
>
	<div class="c-sidebar__content">
		<ng-content></ng-content>
	</div>

	<div class="c-sidebar__actions">
		<ng-content select="actions"></ng-content>
	</div>
</aside>
