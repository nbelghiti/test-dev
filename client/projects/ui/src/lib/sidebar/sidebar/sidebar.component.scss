// =============================================================================
// :: Imports
// =============================================================================
@import "global";

// ============================================================================
// :: Settings
// ============================================================================
$sidebar-width: 34rem;

// ============================================================================
// :: Sidebar component
// ============================================================================
.c-sidebar {
	position: absolute;
	top: 0;
	right: 0;
	z-index: $z-index-sky;

	display: flex;
	flex-direction: column;
	width: $sidebar-width;
	height: 100%;

	background-color: $background-color-base;
	border-left: $border-width-base solid $border-color-base;
}

// ============================================================================
// :: Elements
// ============================================================================
.c-sidebar__content {
	flex-grow: 1;
	overflow-x: hidden;
	overflow-y: auto;
}
