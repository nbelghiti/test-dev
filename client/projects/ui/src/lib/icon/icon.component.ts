import { Component, HostBinding, Input } from '@angular/core';

import { IconName, IconSize, IconType } from './icon.types';

@Component({
	selector: 'ui-icon',
	templateUrl: './icon.component.html',
	styleUrls: ['./icon.component.scss']
})

export class IconComponent {
	@Input() public name: IconName;
	@Input() public type: IconType = null;
	@Input() public size: IconSize = null;

	@HostBinding('class.c-icon') public className = true;
}
