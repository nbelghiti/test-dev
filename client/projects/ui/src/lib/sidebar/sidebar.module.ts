import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { RouterModule } from '@angular/router';

import { IconModule } from '../icon/icon.module';
import { InfoAccordionModule } from '../info-accordion/info-accordion.module';
import { InfoButtonModule } from '../info-button/info-button.module';
import { InfoLinkModule } from '../info-link/info-link.module';
import { LinkModule } from '../link/link.module';
import { RowModule } from '../row/row.module';
import { TeamModule } from '../team/team.module';

import { SidebarComponent } from './sidebar/sidebar.component';

@NgModule({
	imports: [
		CommonModule,
		IconModule,
		InfoAccordionModule,
		InfoLinkModule,
		InfoButtonModule,
		LinkModule,
		RouterModule,
		RowModule,
		TeamModule,
	],
	declarations: [
		SidebarComponent,
	],
	exports: [
		SidebarComponent,
	]
})
export class SidebarModule {}
