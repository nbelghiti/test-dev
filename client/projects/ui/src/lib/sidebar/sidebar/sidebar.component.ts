import { Component, Input } from '@angular/core';

import { IconName } from '../../icon/icon.types';

@Component({
	selector: 'ui-sidebar',
	templateUrl: './sidebar.component.html',
	styleUrls: ['./sidebar.component.scss']
})
export class SidebarComponent {
	@Input() public title: string;

	public IconName: typeof IconName = IconName;
}
