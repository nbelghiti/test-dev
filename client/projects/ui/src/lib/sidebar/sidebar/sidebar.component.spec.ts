import { ComponentFixture, TestBed, async } from '@angular/core/testing';

import { IconComponent } from '../../icon/icon.component';

import { SidebarComponent } from './sidebar.component';

describe('UserInfoComponent', () => {
	let component: SidebarComponent;
	let fixture: ComponentFixture<SidebarComponent>;

	beforeEach(async(() => {
		TestBed.configureTestingModule({
			declarations: [
				SidebarComponent,
				IconComponent,
			],
		})
		.compileComponents();
	}));

	beforeEach(() => {
		fixture = TestBed.createComponent(SidebarComponent);
		component = fixture.componentInstance;
		fixture.detectChanges();
	});

	it('should create', () => {
		expect(component).toBeTruthy();
	});

	it('Should render correctly', () => {
		expect(fixture).toMatchSnapshot();
	});
});
