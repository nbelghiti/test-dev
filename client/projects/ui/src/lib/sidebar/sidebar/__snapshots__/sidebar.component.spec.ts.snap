// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserInfoComponent Should render correctly 1`] = `
<ui-sidebar
  IconName={[Function Object]}
>
  <aside
    class="c-sidebar"
  >
    <div
      class="c-sidebar__content"
    />
    <div
      class="c-sidebar__actions"
    />
  </aside>
</ui-sidebar>
`;
